### PR TITLE
✨ Prevent BMH from transitioning when an invalid BIOS setting is detected

### PR DESCRIPTION
**Pull request by tliu2021**
**What this PR does / why we need it**:
Currently, the installation process ignores invalid BIOS settings, which can result in a system booting without the desired configuration. This commit ensures that the process halts when an invalid BIOS setting is detected, allowing the user to correct the issue before proceeding, thereby preventing unintended configurations.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: controllers/metal3.io/baremetalhost_controller.go [hunk 1]
```diff
@@ -1128,8 +1129,14 @@ func (r *BareMetalHostReconciler) actionPreparing(prov provisioner.Provisioner,
 	hfsDirty, hfs, err := r.getHostFirmwareSettings(info)
 
 	if err != nil {
-		// wait until hostFirmwareSettings are ready
-		return actionContinue{subResourceNotReadyRetryDelay}
+		switch {
+		case errors.As(err, &ConfigurationError{}):
+			// wait for the user to correct the error
+			return actionContinue{configurationErrRetryDelay}
```
Comment (dtantsur) [inline controllers/metal3.io/baremetalhost_controller.go:1135]: Why is it different from what we had before? You're only changing the retry delay for 1 to 1.5 minutes. Did you intend to you something else than actionContinue?

Comment (tliu2021) [reply controllers/metal3.io/baremetalhost_controller.go:1135]: Previously, the operation did not return an error, and there was no retry delay (see line 1892 below).

**Comment by tliu2021** [comment]
This behavior needs to be triggered based on a label in the HFS or BMH.

File: controllers/metal3.io/baremetalhost_controller.go (hunk 1 above)
Comment (dtantsur) [reply controllers/metal3.io/baremetalhost_controller.go:1135]: I think on line 1132 the error would be returned in any case.

**Pull request closed without merging**